extractor.py

```diff
@@ -1,146 +1,186 @@
-            return None
-        return prog + ' ' + self.attrs[ext] + ' ' + filename
+    return _response_builder(None, _result_builder(True, None), 'Did not find a configuration')
 
-    def get_proper_ext(self, extension):
-        if not extension:
-            return None
-        if len(extension) > 2:
-            print("We don't support this extension.")
-            return None
-        if extension[0] is None:
-            if extension[1] is None:
-                print("We don't support this extension.")
-                return None
-            return self.exts[extension[1]]
-        else:
-            if extension[1] is None:
-                return self.exts[extension[0]]
-            _ext0 = self.exts[extension[0]]
-            _ext1 = self.exts[extension[1]]
-            appended_ext = _ext0 + _ext1
-            return appended_ext
-
-    def get_prog(self, _ext):
-        if not _ext:
-            print("Invalid extension")
-            return None
-        _prog = self.progs[_ext]
-        if not _prog:
-            print("We don't support this file type")
-            return None
-        return _prog
+def _create_configuration_file():
+    data = {_configs_str: []}
 
-class Extractor:
+    with open(config.EXTRACTOR_CONFIG_FILE, 'w') as outfile:
+        json.dump(data, outfile, indent=2)
 
-    def get_prog(self, extensions):
-        if not extensions:
-            return None
+def _add_config(extension, encoding, prog, attrs):
+    _logd('add_config called:', extension, encoding, prog, attrs)
 
-    def build_command(self, program, filename):
-        if not program:
-            return None
-        if not filename:
+    json_data = None
+    with open(config.EXTRACTOR_CONFIG_FILE, 'r') as f:
+        try:
+            json_data = json.load(f)
+        except JSONDecodeError:
+            _logd('add_config exception')
+            print('Could not add a new configuration to your file')
             return None
 
-    def guess_type(self, filename):
-        ext = mimetypes.guess_type(filename, False)
-        if not ext:
-            return None
-        if DEBUG:
-            print(ext)
-        return ext
+    item = {
+        _ext_str: extension,
+        _encoding_str: encoding,
+        _prog_str: prog,
+        _attrs_str: attrs
+    }
 
-    def extract(self, filename):
-        print("Will try to extract file:", filename)
-        ext = self.guess_type(filename)
-        if not ext:
-            print("Could not guess filetype. Cannot extract")
-            return None
-        prog = Program()
-        _ext = prog.get_proper_ext(ext)
-        if not _ext:
-            print("Cannot extract at this moment")
-            return None
+    json_data[_configs_str].append(item)
+
+    with open(config.EXTRACTOR_CONFIG_FILE, 'w') as outfile:
+        json.dump(json_data, outfile, indent=2)
+    return item
+
+def _configure_new_item(extension, encoding):
+    prog = input('What\'s the program? (No options) ') or None
```

Comment:
Optionally we could have a default or recommended program in case the user inputs blank

Reply:
I think that this could be a future work